forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabRepository.java

```diff
@@ -635,16 +635,21 @@ private Map<String, Set<Hash>> getCommitTitleToCommitsMap(ReadOnlyRepository loc
                 .param("since", lastCommitTime.format(DateTimeFormatter.ISO_DATE_TIME))
                 .param("all", "true")
                 .execute()
-                .asArray();
+                .asArray()
+                .stream()
+                .toList()
+                .reversed();
+
         for (var commit : commits) {
             var hash = new Hash(commit.get("id").asString());
             var title = commit.get("title").asString();
-            commitTitleToCommits.computeIfAbsent(title, t -> new LinkedHashSet<>()).add(hash);
+            ((LinkedHashSet<Hash>) commitTitleToCommits.computeIfAbsent(title, t -> new LinkedHashSet<>())).addFirst(hash);
```

Comment:
Instead of casting to LinkedHashSet, we can use the type directly in the type declaration for `commitTitleToCommits`. We can even use the new shiny `SequencedSet` interface to avoid specifying the concrete class.

Reply:
Sure